diff --git a/api.js b/api.js
index 350708d75403a7dbb72e1fbaa68aea056dcea651..f76dc2c34837a701bfd3b9477b12b8f11445b109 100644
--- a/api.js
+++ b/api.js
@@ -39271,7 +39271,7 @@ var init_libsql = __esm({
     libSQLCredentials = objectType({
       url: stringType().min(1),
       authToken: stringType().min(1).optional()
-    });
+    }).passthrough();
   }
 });
 
diff --git a/api.mjs b/api.mjs
index 3e2cef2fb5bf7339b9aedff32e121a49e5aa7691..da401169ae78c2c5d17e3f876fe5286842013724 100644
--- a/api.mjs
+++ b/api.mjs
@@ -39276,7 +39276,7 @@ var init_libsql = __esm({
     libSQLCredentials = objectType({
       url: stringType().min(1),
       authToken: stringType().min(1).optional()
-    });
+    }).passthrough();
   }
 });
 
diff --git a/bin.cjs b/bin.cjs
index b35924f2df2c4eca4ebaede3ef730dbc8484c8ff..1a4b6cf5fb4c7c44fb02db68fb7f31f1e72bf350 100755
--- a/bin.cjs
+++ b/bin.cjs
@@ -11768,7 +11768,7 @@ var init_libsql = __esm({
     libSQLCredentials = objectType({
       url: stringType().min(1),
       authToken: stringType().min(1).optional()
-    });
+    }).passthrough();
     printConfigConnectionIssues = (options, command) => {
       let text = `Please provide required params for 'turso' dialect:
 `;
@@ -76806,7 +76806,7 @@ var init_connections = __esm({
         const { createClient } = await import("@libsql/client");
         const { drizzle } = await import("drizzle-orm/libsql");
         const { migrate: migrate2 } = await import("drizzle-orm/libsql/migrator");
-        const client = createClient({
+        const client = credentials2.client ?? createClient({
           url: normaliseSQLiteUrl(credentials2.url, "libsql"),
           authToken: credentials2.authToken
         });
